storage/remote/write.go

```diff
@@ -93,6 +94,7 @@ func NewWriteStorage(logger *slog.Logger, reg prometheus.Registerer, dir string,
 		interner:          newPool(),
 		scraper:           sm,
 		quit:              make(chan struct{}),
+		// TODO: needed?
```

Comment:
needed?
(removing it will allow us to get rid of the whole `timestampTracker` logic)